src/plots/get_data.js

```diff
@@ -45,26 +45,42 @@ exports.getSubplotCalcData = function(calcData, type, subplotId) {
  * the calcdata we *will* plot with this module, and the ones we *won't*
  *
  * @param {array} calcdata: as in gd.calcdata
- * @param {object|string} typeOrModule: the plotting module, or its name
+ * @param {object|string|fn} arg1:
+ *  the plotting module, or its name, or its plot method
  *
  * @return {array[array]} [foundCalcdata, remainingCalcdata]
  */
-exports.getModuleCalcData = function(calcdata, typeOrModule) {
+exports.getModuleCalcData = function(calcdata, arg1) {
     var moduleCalcData = [];
     var remainingCalcData = [];
-    var _module = typeof typeOrModule === 'string' ? Registry.getModule(typeOrModule) : typeOrModule;
-    if(!_module) return [moduleCalcData, calcdata];
+
+    var plotMethod;
+    if(arg1) {
+        var typeOfArg1 = typeof arg1;
```

Comment:
Does a falsy `arg1` happen?

Also, I recall seeing somewhere that js can optimize `typeof arg1 === 'string'` internally so it never has to generate the string values, whereas with `var typeOfArg1 = typeof arg1; typeOfArg1 === 'string'` it can't do that so it's actually better to write out `typeof arg1 === '...'` in each clause. I can't find it now though...

Reply:
Thanks for tips! 

Simplified in -> https://github.com/plotly/plotly.js/pull/2574/commits/e5c860a10488e0ddfdb8eba445909a90b643f3c2